src/core/inductive.ml

```diff
@@ -175,31 +239,43 @@ let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
     let f_not_rec : tbox -> tvar -> tbox = fun a _ -> a in
     let init : tbox = _Type in
     fold_cons_typ pos codom inj_var build_rec_hyp domrec dom ind_sym
-      cons_sym f_rec f_not_rec init x_str
+      cons_sym f_rec f_not_rec init x_str assoc_predicats
   in
-
-  (* STEP 3: Create the conclusion of the inductive principle *)
-  let codom ts =
-    let x = Bindlib.new_var mkfree x_str in
-    let t = Bindlib.bind_var x (prf_of_p ts (_Vari x)) in
-    _Prod (fapp ind_sym ts) t
+  (* Very closed to the function "fold_right2", but "f" is the function
+     "fold_right" and it needs a neutral element. *)
+  let rec clauses ind_typ_list cons_list_list e =
```

Comment:
Again, to avoid potential problems with list lengths, it would be better to take as argument `(sym * sym list) list`.

Reply:
If a day we have a potential problems with list lengths, it would be more difficult to find it if the type of the argument is `(sym * sym list) list`.